1.4/Source/Bullet_Impact_DoubleWind.cs

```diff
@@ -0,0 +1,38 @@
+﻿using HarmonyLib;
+using RimWorld;
+using Verse;
+
+namespace VPE_Ranger
+{
+    [HarmonyPatch(typeof(Bullet))]
+    [HarmonyPatch("Impact")]
+    public class Bullet_Impact_DoubleWind
+    {
+        public static void Postfix(Thing hitThing, Bullet __instance)
+        {
+            if(__instance == null || !(__instance.Launcher is Pawn pawn) || !pawn.health.hediffSet.HasHediff(VPERanger_DefOf.MakaiRanger_BuffFour))
+            {
+                return;
+            }
+            if(__instance.usedTarget.Thing is Pawn pawnEn)
+            {
+                if(!pawnEn.health.hediffSet.HasHediff(VPERanger_DefOf.MakaiRanger_BuffFour_Mark) && pawn.health.hediffSet.HasHediff(VPERanger_DefOf.MakaiRanger_BuffFour_Speed))
+                {
+                    pawn.health.RemoveHediff(pawn.health.hediffSet.GetFirstHediffOfDef(VPERanger_DefOf.MakaiRanger_BuffFour_Speed));                    
+                }
+                pawnEn.health.AddHediff(VPERanger_DefOf.MakaiRanger_BuffFour_Mark);

```

Comment:
this will always add the hediff, don't you want it only adding the hediff if it isn't there?